packages/p2p-v2/src/providers/AdvertiserPaymentMethodsProvider/AdvertiserPaymentMethodsProvider.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { createContext, Dispatch, PropsWithChildren, useContext, useReducer } from 'react';
+import { TAdvertiserPaymentMethodsConfig, TReducerAction } from '../types';
+import advertiserPaymentMethodsReducer from './advertiserPaymentMethodsReducer';
+
+const AdvertiserPaymentMethodsConfigContext = createContext<TAdvertiserPaymentMethodsConfig | null>(null);
+
+const AdvertiserPaymentMethodsConfigDispatchContext = createContext<Dispatch<TReducerAction> | null>(null);
+
+export const AdvertiserPaymentMethodsProvider = ({ children }: PropsWithChildren<unknown>) => {
+    const [advertiserPaymentMethodsConfig, dispatch] = useReducer(advertiserPaymentMethodsReducer, {});
+    return (
+        <AdvertiserPaymentMethodsConfigContext.Provider value={advertiserPaymentMethodsConfig}>
+            <AdvertiserPaymentMethodsConfigDispatchContext.Provider value={dispatch}>
+                {children}
+            </AdvertiserPaymentMethodsConfigDispatchContext.Provider>
+        </AdvertiserPaymentMethodsConfigContext.Provider>
+    );
+};
+
+export const useAdvertiserPaymentMethodsConfig = () => {
```

Comment:
Maybe a suggestion for the namings:
`usePaymentMethods`
`usePaymentMethodsDispatch`
`PaymentMethodsContext`
`PaymentMethodsDispatchContext`